src/laser_geometry.cpp

```diff
@@ -437,7 +437,8 @@ const boost::numeric::ublas::matrix<double>& LaserProjection::getUnitVectors_(do
     {
       //check to see if we want to keep the point
       const float range = scan_in.ranges[i];
-      if ((preservative && std::isfinite(ranges(i,0))) || (range < range_cutoff && range >= scan_in.range_min))
```

Comment:
If we are keeping the NAN in `preservative` mode we might as well keep the -INF also:

`if (preservative || (range < range_cutoff && range >= scan_in.range_min))
`